### DISCUSSION TITLE
Improvement of the auto-escaping feature

### DISCUSSION THREAD

**Discussion by Amaury** [smarty-php/smarty#1031] [Ideas]
Hi,

The auto-escaping feature is neat but it has two drawbacks:

- When using legacy templates with `escape` modifiers, data are escaped twice.
- Once enabled, there is no way to prevent the escaping of an expression.

I sent a Pull Request to improve that: https://github.com/smarty-php/smarty/pull/1030

Looking forward to discussing it.

**Comment by Amaury** [comment]
@wisskid The [documentation website](https://smarty-php.github.io/smarty/stable/) seems not up-to-date with the [5.4.0](https://github.com/smarty-php/smarty/releases/tag/v5.4.0) release.

I don't see the [modifications](https://github.com/smarty-php/smarty/pull/1030/files) in `api/configuring` and `language-modifier-escape` pages, and the new `language-modifier-raw` page doesn't exist.

Is there something to do?
Many thanks in advance!

**Reply by wisskid** [reply]
thanx, forgot to publish the changes! (and did so just now)

**Reply by Amaury** [reply]
Great, thanks!
